pkg/deploy/controller/deploymentconfig/deploymentconfig_controller.go

```diff
@@ -125,6 +148,15 @@ func (c *DeploymentConfigController) Handle(config *deployapi.DeploymentConfig)
 				if err != nil {
 					return err
 				}
+				// We need to make sure we own that RC or adopt it if possible
+				isOurs, err := cm.ClaimReplicationController(rc)
```

Comment:
@kargakis 

> What happens if the cache copy is not yet updated with the owner reference you set above?

This is in `RetryOnConflict` block so in the worst case caches should propagate while doing the backoff, right? We could use an uncached client here.

> What happens if the cache copy is not yet updated with the owner reference you set above? 

I think that the second patch should result in noop (and succeed).

> Not sure about having the RC diverge just a couple of Go code after the initial adoption and this is a cache lookup but maybe this doesn't hurt as a last line of defense?

Well not doing this and patching a RC we don't own due to race condition would break 2. or 3. [controller rule](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/controller-ref.md#the-three-laws-of-controllers)

But that also gets me thinking that we can break it under some conditions. There is still race condition in the patches we use (taken from upstream). I think they should really use optimistic concurrency in form of resourceVersion to be completely correct. Hope that works with `PATCH`. I'll take the discussion upstream to figure it out because the do the same.